src/data/nat/basic.lean

```diff
@@ -1205,6 +1205,9 @@ by rw [←choose_mul_fact_mul_fact hk, mul_assoc]; exact dvd_mul_left _ _
 @[simp] lemma choose_symm {n k : ℕ} (hk : k ≤ n) : choose n (n-k) = choose n k :=
 by rw [choose_eq_fact_div_fact hk, choose_eq_fact_div_fact (sub_le _ _), nat.sub_sub_self hk, mul_comm]
 
+lemma choose_symm' {n a b : ℕ} (h : n = a + b) : nat.choose n a = nat.choose n b :=
+by { convert nat.choose_symm (nat.le_add_left _ _), rw nat.add_sub_cancel}
```

Comment:
How about this statement `nat.choose (a+b) a` = `nat.choose (a+b) b`?

Reply:
I considered this, but when I used this (to prove the Erdos-Ko-Rado theorem here) it was in the context of proving `choose (n - 1) (n - r) = choose (n - 1) (r - 1)` in the context `1 ≤ r ≤ n`, where using your version would be quite a bit more awkward. So, I went with this version instead.

Reply:
Is it that much work. You just have to `rw` with your proof that `n = a + b` first, instead of supplying it as a hypothesis. I guess it could be slightly awkward if you had other occurrences of `n-1` in your goal, but I'm not convinced it's worth changing the statement for the sake of one theorem.

Reply:
This is only here as a convenience statement anyway - isn't it worth having an easier to use result? Also, your version would follow immediately from this one using just `rfl`.

Reply:
There's not a total ordering on convenience here. Let's have both, with names `choose_symm_add` and `choose_symm_of_eq_add`.

Reply:
Should be done!